propagation.go

```diff
@@ -215,6 +215,13 @@ func (p *BinaryPropagator) Inject(
 	return nil
 }
 
+// These are intended to be unimaginably high: above this number we declare
+// the data is corrupted, to protect against running out of memory
+const (
+	maxBinaryBaggage  = 1024 * 1024
+	maxBinaryValueLen = 100 * 1024 * 1024
```

Comment:
I don't think these huge values make sense. W3C baggage is limited to 8k (https://github.com/w3c/baggage/blob/master/baggage/HTTP_HEADER_FORMAT.md#limits).

Reply:
Yeah fair enough. I looked in OpenTracing and didn’t see any recommended sizes or limits. 

Any thoughts on key/value limits?

Reply:
Let's use w3c baggage limits
* Maximum number of name-value pairs: 180.
* Maximum number of bytes per a single name-value pair: 4096.
* Maximum total length of all name-value pairs: 8192.